### DISCUSSION TITLE
Postgres Table Name

### DISCUSSION THREAD

**Discussion by SeeYangZhi** [run-llama/llama_index#14766] [Q&A]
https://github.com/run-llama/llama_index/blob/a65adeaf4834913817e353badaae2fa8374fef83/llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py#L60
Is there a reason the table name must be prefixed with `data_`? Can we allow complete customisation?

**Comment by trevoro** [comment]
I also would like to know the answer to this, or at the very least have this documented directly within the library. Data is one of those variable names that don't mean anything and this convention forces people to go around what they might already have in their existing pg database. If I steel man the existing implementation, my only guess would be that this is desired b/c of the auto-table generation behaviour, but it shouldn't be necessary. We should be able to trust developers to chose non-overlapping table names.

**Comment by logan-markewich** [comment]
I honestly don't know why its here either, but since its there now, removing it needs to be done with care -- handling backwards compatibility

I welcome a PR

**Comment by eon01** [comment]
I also ran into the `data_` prefix issue when integrating LlamaIndex with an existing PostgreSQL schema.

This creates confusion (`PGVectorStoretable_name` exposes only the logical name not the one with `data_`) and can break raw SQL operations unless you manually reconstruct the full physical name. It also limits integration with naming conventions (e.g., when you're forced to use specific table names for compliance, migrations, or org standards..etc).

It would great if users can optionally override the prefix via `PGVectorStore.from_params(..., table_prefix="...")` — defaulting to `"data_"` for backwards compatibility.

**Reply by logan-markewich** [reply]
Feel free to make a PR!
